### DISCUSSION TITLE
function from_data in json.py shoud handle class serializable

### DISCUSSION THREAD

**Discussion by eggqq007** [Textualize/rich#1534] [General]
We consider to consume function from_data in json.py to print prettify json. But I found it doesn't accept 'default' which just like default in json.dumps.  It means it may hit issue while converting class into json.
----------------------------------------------------------------------------------
json.dump(obj, fp, *, skipkeys=False, ensure_ascii=True, check_circular=True, allow_nan=True, cls=None, indent=None, separators=None, default=None, sort_keys=False, **kw)

If specified, default should be a function that gets called for objects that can’t otherwise be serialized. It should return a JSON encodable version of the object or raise a TypeError. If not specified, TypeError is raised.

**Comment by skshetry** [comment]
Support for `default` was added in #1547.

**Reply by eggqq007** [reply]
> Support for `default` was added in #1547.

Thanks for working on it
